Added tpause C0.1 and C0.2 as two separate kernels. Updated build.sh

Sinduri11 opened the pull request.

aayasin reviewed (changes requested):
Sinduri,
Thanks for contributing the TPAUSE kernel.
I did code review. Key points:
1. Argument aren't user friendly. In your code, it requires the user to know current TSC value to be able to calculate a future target one. Specifying a TSC delta is way better, 
2. Code duplication. The C0 sub-state can be made a separate arg. This help maintaining the code long term.
3. Besides to #1, another cons, is the SUB and ADD in while condition and body make the loop lesser pure TPAUSE.

I coded fixes of the two in the attached, and tested it is working. Feel free to reuse and re-submit.
[tpause.c.txt](https://github.com/aayasin/perf-tools/files/11724645/tpause.c.txt)

Ahmad

Sinduri11 commented:
Thank you for the feedback Ahmad. I've incorporated the suggested changes - deleted the 2 separate c files and added tpause.c. Updated the build file to reflect the same.
Please let me know if anything else is needed.

thanks,
Sinduri

aayasin reviewed (approved):
Thanks for incorporating the fixes
